File: programmers/python/level1/p77484.py
def solution(lottos, win_nums):
    count = 0
    zero_count = 0
    
    for l in lottos:
        if l == 0:
            zero_count += 1
        elif l in win_nums:
            count += 1
    
    good = 1 if (count == 0 and zero_count == 0) else count + zero_count
    bad = 1 if count == 0 else count
    
    return [7 - good, 7 - bad]

l, w = [0, 0, 0, 0, 0, 0], [38, 19, 20, 40, 15, 25]

print(solution(l, w))

File: book/java/src/y25/level1/P77484.java
package y25.level1;
import java.util.*;

public class P77484 {
    public int[] solution(int[] lottos, int[] win_nums) {
        int zeroCount = 0;
        List<Integer> winSet = new ArrayList<>();

        for (int num : win_nums) {
            winSet.add(num);
        }

        for (int lotto: lottos) {
            winSet.remove(Integer.valueOf(lotto));
            if (lotto == 0) {
                zeroCount++;
            }
        }

        int row = winSet.size() != 6 ? winSet.size() + 1 : winSet.size();

        if (zeroCount > 0) {
            winSet.subList(0, zeroCount).clear();
        }

        int high = winSet.size() == 6 ? 6 : 7 - (6 - winSet.size());

        return new int[]{high, row};
    }
}
